### ISSUE TITLE
Long, unbroken link text (such as email addresses) can break out of page container

### ISSUE THREAD

**Issue by abbott567** [alphagov/govuk-frontend#2233]
## Summary
The standard wrapping on body content can create a scenario where long email addresses can cause a service to fail [WCAG 2.1 - 1.4.10 Reflow](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html).

Reflow states that content must not force horizontal scrolling in a viewport of 320px wide unless it's a map, a table, a game or something else where it is necessary. For regular content it needs to reflow and maintain vertical scrolling only.

If you provide an email address which does not contain regular breaking characters such as hyphens, then the email address refuses to break and forces the viewport to horizontal scroll.

By adding the CSS property: `overflow-wrap:anywhere` the issue is resolved.

## Steps to reproduce the issue
1. Add a long email address to some body content which does not contain any standard breaking characters. For example:
```html
<a href="#" class="govuk-link">
  anExampleOfAReallyLongEmailAddress@WhichPushesTheContentOver320PixelsWide.com
</a>
```

2. Reduce the viewport width to 320px wide and observe horizontal scrolling.

![image](https://user-images.githubusercontent.com/9450064/119004428-7a16c180-b986-11eb-9639-678c87081440.png)

## Environment (where applicable)
- Operating system: OS X Catalina + Windows 10
- Browser: Chrome (Mac), Chrome (Win), Edge, Safari
- GOV.UK Frontend Version: 3.12.0

**Comment by abbott567** [comment]
Worth pointing out the service this was observed on was a staff facing system, so the email address was probably longer than usual. It was chaining together team names and business areas using full stops. For example: `jonathan.macfarland@orangeteam.improvementprojects.department.gov.uk

**Comment by 36degrees** [comment]
I'm not sure there's an easy generic fix for this. Adding `overflow-wrap:anywhere` to the body (or applying it generally to paragraphs, links etc) causes issues wherever we're relying on intrinsic sizing for layout, for example in tables or the phase banner.

![Screenshot 2021-06-03 at 15 05 12](https://user-images.githubusercontent.com/121939/120658604-7c484800-c47d-11eb-9881-aae406b24c87.png)
![Screenshot 2021-06-03 at 15 06 40](https://user-images.githubusercontent.com/121939/120658608-7ce0de80-c47d-11eb-8162-8f84df2001d3.png)

I think this is something that would need to be handled on a case by case basis, like we have for summary list.

**Comment by abbott567** [comment]
Yeah, I guess even if it was available as a class so you could override it manually. Possibly like `.govuk-!-force-wrap` or something.
